src/overlays/actors/ovl_En_Fall/z_en_fall.c

```diff
@@ -36,75 +64,820 @@ const ActorInit En_Fall_InitVars = {
+        this->actor.draw == NULL) {
+        EnFall_MoonsTear_Initialize(this);
+    }
+
+    if (this->actor.draw != NULL) {
+        if (Math_Vec3f_StepTo(&this->actor.world.pos, &this->actor.home.pos, this->actor.speedXZ) <= 0.0f) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_GORON_BOUND_1);
+            gSaveContext.weekEventReg[0x4A] |= 0x80;
+            gSaveContext.weekEventReg[0x4A] |= 0x20;
+            Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_TEST, this->actor.world.pos.x,
+                               this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, -2);
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_CLEAR_TAG, this->actor.world.pos.x,
+                        this->actor.world.pos.y, this->actor.world.pos.z, 0, 0, 0, CLEAR_TAG_LARGE_EXPLOSION);
+            this->actor.draw = NULL;
+            this->actionFunc = EnFall_DoNothing;
+        } else {
+            func_800B9010(&this->actor, NA_SE_EV_MOONSTONE_FALL - SFX_FLAG);
+        }
+    }
+}
+
+void EnFall_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnFall* this = THIS;
+    this->actionFunc(this, globalCtx);
+}
+
+/**
+ * Updates the alpha for every vertex in the fire ball so that some parts of
+ * the sphere are more transparent or opaque than others.
+ */
+void EnFall_FireBall_SetPerVertexAlpha(f32 fireBallAlpha) {
+    static u8 sAlphaTableIndex[] = { 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 0, 1, 1, 1, 4, 0, 4, 0, 1, 1, 1, 3, 0,
+                                     3, 0, 1, 1, 1, 4, 4, 1, 1, 0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0,
+                                     1, 1, 1, 1, 1, 4, 0, 4, 4, 0, 4, 4, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 3, 3, 0, 0, 1,
+                                     1, 1, 1, 1, 1, 4, 4, 0, 4, 0, 1, 1, 1, 3, 0, 3, 3, 0, 0, 1, 1, 1, 1, 1, 1, 4, 4,
+                                     0, 4, 4, 0, 1, 1, 1, 1, 1, 1, 3, 3, 0, 0, 0, 0, 0, 2, 2, 1, 1, 0, 0, 2, 2, 0, 1,
+                                     1, 0, 0, 1, 0, 2, 0, 0, 1, 0, 2, 0, 0, 2, 1, 2, 0, 1, 0, 1, 0, 0, 1, 2, 2, 0, 0,
+                                     2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0, 0, 0, 2, 0, 2, 2, 0, 1, 1, 0, 0, 1, 0, 0,
+                                     1, 2, 2, 0, 0, 0, 2, 2, 1, 1, 0, 0, 1, 1, 0, 0, 2, 2, 0, 0 };
+
+    s32 pad;
+    u8 perVertexAlphaTable[5];
+    Vtx* vertices;
+    s32 i;
+
+    vertices = Lib_SegmentedToVirtual(&D_060004C0);
+    if (fireBallAlpha > 1.0f) {
+        fireBallAlpha = 1.0f;
+    }
+    perVertexAlphaTable[0] = 0;
+    perVertexAlphaTable[1] = (s8)(255.0f * fireBallAlpha);
+    perVertexAlphaTable[2] = (s8)(155.0f * fireBallAlpha);
+    perVertexAlphaTable[3] = (s8)(104.0f * fireBallAlpha);
+    perVertexAlphaTable[4] = (s8)(54.0f * fireBallAlpha);
+
+    for (i = 0; i < 209; i++, vertices++) {
+        vertices->v.cn[3] = perVertexAlphaTable[sAlphaTableIndex[i]];
+    }
```

Comment:
```suggestion
    for (i = 0; i < ARRAY_COUNT(sAlphaTableIndex); i++, vertices++) {
        vertices->v.cn[3] = perVertexAlphaTable[sAlphaTableIndex[i]];
    }
```
I can't tell by the size of the table just by looking at it, but I would guess it would be pretty close?

Reply:
No, that's right. Good catch, thanks!